hilti/toolchain/src/compiler/codegen/expressions.cc

```diff
@@ -32,6 +32,32 @@ struct Visitor : hilti::visitor::PreOrder<std::string, Visitor> {
         return fmt("%s = %s", cg->compile(n.target(), true), cg->compile(n.source()));
     }
 
+    result_t operator()(const expression::BuiltinFunction& n) {
+        // We use a statement expression (this is an extension supported by
+        // both GCC and Clang) in order for this to be callable in global
+        // contexts.
+        // This "obvious" approach runs into issues there since temporaries
+        // potentially created via `CodeGen::compile` require a block which is
+        // not present for certain globals:
```

Comment:
I'm still wondering why this isn't working and/or whether we'll run into the same problem elsewhere when initializing globals. But we can see, this will work for now.

Reply:
I have filed #748 for this issue (putting more data of `BuiltinFunction` nodes in the AST did not help this issue).